app/controllers/users_controller.rb

```diff
@@ -0,0 +1,27 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      flash[:success] = t(".success")
+      redirect_to @user
+    else
+      flash[:danger] = t(".error")
+      render :new
+    end
+  end
+
+  def show
+    @user = User.find_by(params[:id])
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:name, :email, :password, :password_confirmation)
```

Comment:
```suggestion
    params.require(:user).permit User::USER_PARAMS
```
- ở model user Thêm biến constant USER_PARAMS = %i(name email password password_confirmation).freeze
- giải thích method permit xem sử dụng ntn và vì sao phải sử dụng nhỉ ?

Reply:
permit nhằm khi tạo user mới chỉ có những giá trị như name, email, password, password_confirmation.
Được sử dụng trong StrongParameters để tránh việc hacker có thể thêm dữ liệu vào khi ta đang tạo một người dùng mới, ví dụ admin: true thì user sẽ trở thành admin